scripts/edgepy.py

```diff
@@ -37,39 +59,177 @@ def __init__(self, args):
             print(self.dge_list)
 
         elif args.mongo_config:
+            # This section is only useful for MongoDB based analyses.  Talk to @apfejes about this section if you have
+            # any questions.
+
             config = configparser.ConfigParser()
             config.read(args.mongo_config)
 
+            if args.group1_sample_names and args.group2_sample_names:
+                key_name = 'sample_name'
+                key_value = args.group1_sample_names + args.group2_sample_names
+
+            elif args.key_name and args.mongo_key_value:
+                key_name = args.mongo_key_name
+                key_value = args.mongo_key_value
+            else:
+                raise ValueError("Insufficient parameters for use of Mongodb")
+
             mongo_importer = ImportFromMongodb(
                 host=config.get("Mongo", "host"),
                 port=config.get("Mongo", "port"),
-                mongo_key_name=args.mongo_key_name,
-                mongo_key_value=args.mongo_key_value,
+                mongo_key_name=key_name,
+                mongo_key_value=key_value,
```

Comment:
Why `key_name` and `key_value` and not just `key` and `value` like how people think about dictionaries?

Reply:
thanks - good point!